Monika After Story/game/script-windowreacts.rpy

```diff
@@ -0,0 +1,556 @@
+
+label mas_wrs_pinterest:
+    $ wrs_success = mas_display_notif(
+        m_name,
+        [
+            "Anything new today, [player]?",
+            "Anything interesting, [player]?",
+            "See anything you like?"
+        ],
+        'Window Reactions'
+    )
+
+    #Unlock again if we failed
+    if not wrs_success:
+        $ mas_unlockFailedWRS('mas_wrs_pinterest')
+    return
+
+init 5 python:
+    addEvent(
+        Event(
+            persistent._mas_windowreacts_database,
+            eventlabel="mas_wrs_duolingo",
+            category=['duolingo'],
+            rules={
+                "notif-group": "Window Reactions",
+                "skip alert": None,
+                "keep_idle_exp": None
+            },
+            show_in_idle=True
+        ),
+        code="WRS"
+    )
+
+label mas_wrs_duolingo:
+    $ wrs_success = mas_display_notif(
+        m_name,
+        [
+            "Learning new ways to say 'I love you,' [player]?",
+            "Learning a new language, [player]?",
+            "What language are you learning, [player]?"
+        ],
+        'Window Reactions'
+    )
+
+    #Unlock again if we failed
+    if not wrs_success:
+        $ mas_unlockFailedWRS('mas_wrs_duolingo')
+    return
+
+init 5 python:
+    addEvent(
+        Event(
+            persistent._mas_windowreacts_database,
+            eventlabel="mas_wrs_wikipedia",
+            category=['wikipedia'],
+            rules={
+                "notif-group": "Window Reactions",
+                "skip alert": None,
+                "keep_idle_exp": None
+            },
+            show_in_idle=True
+        ),
+        code="WRS"
+    )
+
+label mas_wrs_wikipedia:
+    $ wikipedia_reacts = [
+        "Learning something new, [player]?",
+        "Doing a bit of research, [player]?"
+    ]
+
+    #Items in here will get the wiki article you're looking at for reacts.
+    python:
+        wind_name = mas_getActiveWindow(friendly=True)
+        try:
+            cutoff_index = wind_name.index(" - Wikipedia")
+
+            #If we're still here, we didn't value error
+            #Now we get the article
+            wiki_article = wind_name[:cutoff_index]
+
+            # May contain clarification in trailing parentheses
+            wiki_article = re.sub("\\s*\\(.+\\)$", "", wiki_article)
+            wikipedia_reacts.append(renpy.substitute("'[wiki_article]'...\nSeems interesting, [player]."))
+
+        except ValueError:
+            pass
+
+    $ wrs_success = mas_display_notif(
+        m_name,
+        wikipedia_reacts,
+        'Window Reactions'
+    )
+
+    #Unlock again if we failed
+    if not wrs_success:
+        $ mas_unlockFailedWRS('mas_wrs_wikipedia')
+    return
+
+init 5 python:
+    addEvent(
+        Event(
+            persistent._mas_windowreacts_database,
+            eventlabel="mas_wrs_virtualpiano",
+            category=["virtual", "piano"],
+            rules={"notif-group": "Window Reactions", "skip alert": None},
```

Comment:
This one is missing a rule, can add it in your PR since you move wrs anyway.
```suggestion
            rules={
                "notif-group": "Window Reactions",
                "skip alert": None,
                "keep_idle_exp": None
            },
```